src/cloudStorage/controller.js

```diff
@@ -19,21 +23,39 @@ async function generateSignedUploadURL(req, res) {
     async () => {
       let { location } = req.body;
 
-      const options = {
-        version: "v4",
-        action: "write",
-        expires: Date.now() + 15 * 60 * 1000, // 15 minutes
-      };
-
-      // Get a v4 signed URL for uploading file
-      const [url] = await storage
-        .bucket(CLOUD_STORAGE_BUCKET_NAME)
-        .file(location)
-        .getSignedUrl(options);
-
-      return { data: { url } };
+      return { data: { url: await generateSignedURL("write", location) } };
     },
     res,
     "fetching packing slips"
   );
 }
+
+async function deleteCloudStorageObject(filepath) {
+  const deleteOptions = {
+    ifGenerationMatch: generationMatchPrecondition,
```

Comment:
`generationMatchPrecondition` doesnt seem to be defined anywhere

Reply:
This is no longer here